### PR TITLE
[structured config] Fix usage with Assets

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Since asset inputs flow through a separate code path, the `config` parameter wasn't being stripped out properly when using structured config with assets. Fixes + adds test.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11327** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11327" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11327?utm_source=stack-comment).

**Review by schrockn — APPROVED** [review]

File: python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py [hunk 1]
```diff
@@ -136,6 +137,28 @@ def my_asset(config: AnAssetConfig):
...
+        build_assets_job(
+            "blah",
+            [two_assets],
+            config={"ops": {"two_assets": {"config": {"a_string": "foo", "an_int": 2}}}},
+        )
+        .execute_in_process()
+        .success
+    )
+
+
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py:161]: One thing I like to do in test cases like these is to absolutely ensure that the function body has been executed. 

Meaning before the function (in this case the asset):

`executed = {}`

Then in the function body:

`executed["yes"] = True`

Then after job execution:

`assert executed["yes"]`

When the next engineer comes along and they see that they have high confidence that the code is actually executed. Whenever I work on a framework that is calling back into user code that is doing asserts (and not verifying outputs) I usually do this. More than once I've been bitten debugging something only realizing after some time that some function was actually never called.

Don't feel super strongly so I'm not going to block commit, but I generally find it's a good best practice.

**Review by schrockn — APPROVED** [review]
Approved but consider final comment.

File: python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py (hunk 1 above)
Comment (benpankow) [reply python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py:161]: updated

**Pull request merged**